app/services/taxons_to_legacy_associations_tagging.rb

```diff
@@ -0,0 +1,61 @@
+class TaxonsToLegacyAssociationsTagging
+  def call(edition:, user:, selected_taxons:)
+    # Only perform the taxon based tagging for content with no
+    # existing legacy associations, to avoid overriding human tagging
+    # decisions
+    return false if any_legacy_associations?(edition)
+
+    all_legacy_associations = legacy_mapping_for_taxons(selected_taxons)
```

Comment:
It's a small thing, but I'd rename `all_legacy_associations` and the `any_legacy_associations?` function to make it clear what the difference between "any" and "all" is. 

So, any_legacy_associations? could become something like `any_legacy_associations_in_publishing_api?` and all_legacy_associations could be `any_local_legacy_associations`.

Reply:
Apart from this, everything else looks good. 👍